src/pages/items-page/components/Pagination.jsx

```diff
@@ -0,0 +1,72 @@
+import styled from "@emotion/styled";
+import LeftArrowIcon from "@assets/icons/leftArrow";
+import RightArrowIcon from "@assets/icons/rightArrow";
+
+const Pagination = ({ totalCount, pageSize, page, onChange }) => {
+  const totalPage = Math.ceil(totalCount / pageSize);
+  const maxVisible = 5;
+  const half = Math.floor(maxVisible / 2);
+
+  let startPage = page - half;
+  let endPage = page + half;
+
+  if (startPage < 1) {
+    startPage = 1;
+    endPage = Math.min(totalPage, maxVisible);
+  }
+
+  if (endPage > totalPage) {
+    endPage = totalPage;
+    startPage = Math.max(1, totalPage - maxVisible + 1);
+  }
+
+  const tagList = [];
+  for (let i = startPage; i <= endPage; i++) {
+    tagList.push(
+      <PageItem key={i} onClick={() => onChange(i)} active={i === page}>
+        {i}
+      </PageItem>
+    );
+  }
+
+  return (
+    <PaginationList>
+      <PageItem key="first" onClick={() => onChange(1)} active={false}>
+        <LeftArrowIcon />
+      </PageItem>
```

Comment:
❗️ 수정요청
이런 클릭 가능한 요소의 경우 버튼을 사용해주세요. 더 적절한 태그가 있는데 이미지 태그에 onClick 을 주게되면 button에게 기대하는 동작을 제대로 수행하지 못합니다!

```suggestion
      <PageItem onClick={() => onChange(1)} active={false}> // button태그로 바꿔주세요. key 속성 불필요하니 지워주세요
        <LeftArrowIcon />
      </PageItem>
```

추가로 현재 첫 페이지와 마지막 페이지로 이동하는 버튼은 항상 활성화되어 있습니다! 사용자가 이미 첫 페이지나 마지막 페이지에 있을 때 해당 버튼을 disabled 처리하시는 것을 추천드려요.